src/v/cluster/remote_topic_configuration_source.cc

```diff
@@ -97,22 +63,19 @@ remote_topic_configuration_source::set_remote_properties_in_config(
   ss::abort_source& as) {
     cloud_storage::topic_manifest manifest;
 
-    auto [res, key] = co_await download_topic_manifest(
+    auto res = co_await download_topic_manifest(
       _remote, cfg, manifest, bucket, as);
     if (res != errc::success) {
         co_return res;
     }
 
     if (!manifest.get_topic_config()) {
-        vlog(
-          clusterlog.warn,
-          "Topic manifest {} doesn't contain topic config",
-          key);
         co_return errc::topic_operation_error;
     } else {
+        const auto& dl_cfg = manifest.get_topic_config();
         cfg.cfg.properties.remote_topic_properties = remote_topic_properties(
-          manifest.get_revision(),
-          manifest.get_topic_config()->partition_count);
+          manifest.get_revision(), dl_cfg->partition_count);
+        cfg.cfg.properties.remote_label = dl_cfg->properties.remote_label;
```

Comment:
Yes, this is what is being used to create the immutable state -- we must get it from remote storage if it exists. Callers up until now will not know the exact label to use, at least for read replicas and recovery topics. Once its set here, it will then be immutable